src/macaron/database/database_manager.py

```diff
@@ -1,112 +1,136 @@
-# Copyright (c) 2022 - 2022, Oracle and/or its affiliates. All rights reserved.
+# Copyright (c) 2022 - 2023, Oracle and/or its affiliates. All rights reserved.
 # Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
 
 """This DatabaseManager module handles the sqlite database connection."""
-
 import logging
-import sqlite3
+from types import TracebackType
+from typing import Any, Optional
+
+import sqlalchemy.exc
+from sqlalchemy import Table, create_engine, insert, select
+from sqlalchemy.orm import Session, declarative_base
+
+from macaron.database.views import create_view
 
 logger: logging.Logger = logging.getLogger(__name__)
 
+ORMBase = declarative_base()
+
 
 class DatabaseManager:
-    """This class handles and manages the connection to sqlite database during the search session."""
+    """
+    This class handles and manages the connection to sqlite database during the session.
+
+    Note that since SQLAlchemy lazy-loads the fields of mapped ORM objects, if the database connection is closed any
+    orm-mapped objects will become invalid. As such the lifetime of the database manager must be longer than any of the
+    objects added to the database (using add() or add_and_commit()).
+    """
 
-    def __init__(self, db_path: str):
+    def __init__(self, db_path: str, base=ORMBase):  # type: ignore
```

Comment:
I did this to allow multiple bases to exist in the same program, in case we are using multiple databases. It is also so that in the tests we can more carefully control what is added to the base and we don't automatically register the ORM tables if we want to import and test some other functionality.